ssl/tls1_clnt.c

```diff
@@ -392,8 +392,9 @@ static int process_server_hello(SSL *ssl)
         }
     }
 
-    ssl->dc->bm_proc_index = offset;
-    PARANOIA_CHECK(pkt_size, offset);
+    ssl->dc->bm_proc_index = pkt_size;
+    /* This check not always valid w/padding: */
+    /* PARANOIA_CHECK(pkt_size, offset); */
```

Comment:
@earlephilhower That change does not seem correct to me. I read carefully the protocol and there it says that if there is a padding, then it will come in the form of an extension (0x0015: extension type=padding), which will have length and value. So in total the packet size must match the offset. If it does not happen then I would assume that the remote server is misbehaving. Can you paste here a hex dump of such a packet? How can I reproduce the issue ?

Reply:
Sorry, but I don't have a hex dump of a server that did this combining of units, but I think the websites referenced in https://github.com/esp8266/Arduino/issues/3661 show that behavior in combining messages.  I did look at the packets sent out to debug this issue, so I did see it coming across that way.

This is from memory, so apologies if it's a bit hazy:

IIRC the actual message size said one thing (which was I **think** 4-byte aligned length) whereas the message contents themselves were less than this.  I think the unused trailing parts of the message length were filled with 0xff.  Basically the server wanted to word-align messages and not byte-pack them one after the other.  I read the spec and it seems like it's legal (or at least not specifically disallowed) as long as they are ignored and not parsed in any way to change the message interpretation.

FWIW, I believe BearSSL does the same "advance pointer to messagestart+messagelen" and not "advance pointer to amount of data I've processed in this message" bit.  It didn't have an issue with the mentioned websites, and I do believe it is a much more paranoid TLS implementation.